src/affine_dimensions.jl

```diff
@@ -0,0 +1,466 @@
+
+#Affine dimensions from other affine dimensions =========================================================================================================
+function AffineDimensions{R}(s::Real, o::Real, dims::AbstractAffineDimensions, sym::Symbol=:nothing) where {R}
+    new_s = s*scale(dims)
+    new_o = offset(dims) + o*scale(dims) #Scale of o is assumed to be scale of base dimensions
+    return AffineDimensions{R}(new_s, new_o, basedim(dims), sym)
+end
+
+
+function AffineDimensions{R}(s::Real, o::UnionAbstractQuantity, dims::AbstractAffineDimensions, sym::Symbol=:nothing) where {R}
+    new_s = s*scale(dims)
+    new_o = offset(dims) + ustrip(siunits(o)) #Offset is always in SI units
+    return AffineDimensions{R}(new_s, new_o, basedim(dims), sym)
+end
+
+function AffineDimensions{R}(s::Real, o::UnionAbstractQuantity, dims::Dimensions, sym::Symbol=:nothing) where {R}
+    return AffineDimensions{R}(s, ustrip(siunits(o)), dims, sym)
+end
+
+#Affine dimensions from quantities =========================================================================================================================
+function AffineDimensions{R}(s::Real, o::UnionAbstractQuantity, q::UnionAbstractQuantity, sym::Symbol=:nothing) where R
+    q0   = siunits(0*q) #Origin point in SI units
+    oΔ   = siunits(o) - siunits(0*o) #Offset is a difference in affine units
+    dimension(q0) == dimension(oΔ) || throw(DimensionError(o, q)) #Check the units and give an informative error
+    
+    #Obtain SI units of the scale and offset
+    o_si = oΔ + q0 #Total offset is origin plus the offset
+    q_si = siunits(q) - q0 #The scaling quantity must remove the origin
+    
+    #Call the SI quantity constructor
+    return AffineDimensions{R}(s, o_si, q_si, sym)
+end
+
+#Base case when everyting is convrted to si units (offset is assumed to be in SI units)
+function AffineDimensions{R}(s::Real, o::UnionAbstractQuantity{<:Any,<:Dimensions}, q::UnionAbstractQuantity{<:Any,<:Dimensions}, sym::Symbol=:nothing) where R
+    dimension(o) == dimension(q) || throw(DimensionError(o, q))
+    o_val = ustrip(o)
+    q_val = ustrip(q)
+    return AffineDimensions{R}(s*q_val, o_val, dimension(q), sym)
+end
+
+#If a quantity is used only for the dimension, the offset is assumed to be in the same scale as the quantity 
+function AffineDimensions{R}(s::Real, o::Real, q::Q, sym::Symbol=:nothing) where {R, Q<:UnionAbstractQuantity}
+    return AffineDimensions{R}(s, o*q, q, sym)
+end
+
+
+scale(d::AffineDimensions)  = d.scale
+offset(d::AffineDimensions) = d.offset
+basedim(d::AffineDimensions) = d.basedim
+
+with_type_parameters(::Type{<:AffineDimensions}, ::Type{R}) where {R} = AffineDimensions{R}
+constructorof(::Type{AffineDimensions}) = AffineDimensions{DEFAULT_DIM_BASE_TYPE}
+constructorof(::Type{AffineDimensions{R}}) where R = AffineDimensions{R}
+
+function Base.show(io::IO, d::AbstractAffineDimensions)
+    addsign = ifelse(offset(d)<0, "-" , "+")
```

Comment:
Why is this needed? I see `abs(offset(d))` is printed already. Why not just print `offset(d)` directly?